### ISSUE TITLE
Creating an anonymous instance of an annotation does not work

### ISSUE THREAD

**Issue by Sandared** [eclipse/xtext-xtend#205]
I'm trying to create an anomymous instance of my annotation like this:

```
annotation MyConfiguration{
	String name = 'osgi'
}

class InstanceTest{

    def doSomething(){
        
        var MyConfiguration config = new MyConfiguration(){
             
             override annotationType(){
                 null
             }

             override name(){
                 null
             }
        }
    }
}

```
where Xtend complains about not being able to resolve MyConfiguration, although I imported it.

In plain Java everything works fine:

```
public @interface MyConfiguration {
  public String name() default "osgi";
}


public class InstanceTest{
  
  public MyConfiguration doSomething() {
	  MyConfiguration config = new MyConfiguration() {
		
		@Override
		public Class<? extends Annotation> annotationType() {
			return null;
		}
		
		@Override
		public String name() {
			return null;
		}
	};
  }
}

```

**Comment by cdietrich** [comment]
@szarnekow 

i wonder if the else branch of `org.eclipse.xtend.core.typesystem.XtendReentrantTypeResolver.doPrepareLocalTypes(ResolvedTypes, IFeatureScopeSession, JvmFeature, Map<JvmIdentifiableElement, ResolvedTypes>)`
should be adapted as follows

```
		if(type.eClass() == TypesPackage.Literals.JVM_GENERIC_TYPE && ((JvmGenericType) type).isInterface()
								|| type.eClass() == TypesPackage.Literals.JVM_ANNOTATION_TYPE	) {
```

**Comment by szarnekow** [comment]
Yes, that would be good starting point for further tests. Not sure if that is sufficient though.

**Comment by cdietrich** [comment]
it at least compiled with the example given above

**Comment by szarnekow** [comment]
Is it possible to do Intf extends SomeAnno in java?

**Comment by cdietrich** [comment]
gives in JDT 

The annotation type MyAnno should not be used as a superinterface for MyIf

no message in javac

**Comment by cdietrich** [comment]
created https://github.com/eclipse/xtext-xtend/pull/526

**Comment by cdietrich** [comment]
Fixed in 2.15
